aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,106 @@
+The [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configureservices) method, which is optional, is called by the web host before the `Configure` method to configure the app's services. The web host may configure some services before `Startup` methods are called (details are provided in the [Hosting](xref:fundamentals/hosting) topic). By convention, [Configuration options](xref:fundamentals/configuration/index) are set in the `ConfigureServices` method. Adding services to the service container makes them available within the app via [dependency injection](xref:fundamentals/dependency-injection).
 
-For features that require substantial setup there are `Add[Service]` extension methods on [IServiceCollection](https://docs.microsoft.com/aspnet/core/api/microsoft.extensions.dependencyinjection.iservicecollection). This example from the default web site template configures the app to use services for Entity Framework, Identity, and MVC:
+For features that require substantial setup, there are `Add[Service]` extension methods on [IServiceCollection](/dotnet/api/Microsoft.Extensions.DependencyInjection.IServiceCollection). A typical web app registers services for Entity Framework, Identity, and MVC:
 
 [!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
 
-Adding services to the services container makes them available within your application via [dependency injection](xref:fundamentals/dependency-injection).
-
-## Services Available in Startup
-
-ASP.NET Core dependency injection provides services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter (but any registered service can be retrieved from this collection, so additional parameters are not necessary).
+## Services available in Startup
 
-Below are some of the services typically requested by `Startup` methods:
+ASP.NET Core [dependency injection](xref:fundamentals/dependency-injection) provides services during an app's startup. These services are requested by including the appropriate interface as a parameter on the `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter. Any registered service can be retrieved from this collection, so additional parameters aren't necessary.
```

Comment:
This could be clearer. The host provides some services that are available to the Startup class constructor. The app adds additional services via ConfigureServices. Both the host and app services are then available in Configure or later in the application.